bouncer/shared/boost.ts

```diff
@@ -0,0 +1,213 @@
+  });
+  return extrinsicResult;
+}
+
+/// Stops boosting for the given boost pool tier and returns the StoppedBoosting event.
+export async function stopBoosting(
+  asset: Asset,
+  boostTier: number,
+  lpUri = '//LP_BOOST',
+  errorOnFail: boolean = true,
+): Promise<Event | undefined> {
+  await using chainflip = await getChainflipApi();
+  const lp = keyring.createFromUri(lpUri);
+
+  assert(boostTier > 0, 'Boost fee must be greater than 0');
+
+  const observeStoppedBoosting = observeEvent(
+    chainFromAsset(asset).toLowerCase() + 'IngressEgress:StoppedBoosting',
+    chainflip,
+    (event) =>
+      event.data.boosterId === lp.address &&
+      event.data.boostPool.asset === asset &&
+      event.data.boostPool.tier === boostTier.toString(),
+  );
+
+  const extrinsicResult = await lpSubmitExtrinsic(
+    chainflip.tx[ingressEgressPalletForChain(chainFromAsset(asset))].stopBoosting(
+      shortChainFromAsset(asset).toUpperCase(),
+      boostTier,
+    ),
+    lp,
+    errorOnFail,
+  );
+  if (!extrinsicResult.dispatchError) {
+    console.log('waiting for stop boosting event');
+    return observeStoppedBoosting;
+  }
+  console.log('Already stopped boosting');
+  return undefined;
+}
+
+/// Provides liquidity, then adds the funds to the boost pool of the given tier and returns the BoostFundsAdded event.
+export async function addBoostFunds(
+  asset: Asset,
+  boostTier: number,
+  amount: number,
+  lpUri = '//LP_BOOST',
+): Promise<Event> {
+  await using chainflip = await getChainflipApi();
+  const lp = keyring.createFromUri(lpUri);
+
+  assert(boostTier > 0, 'Boost fee must be greater than 0');
+
+  await provideLiquidity(asset, amount, true, lpUri);
+
+  const observeBoostFundsAdded = observeEvent(
+    chainFromAsset(asset).toLowerCase() + 'IngressEgress:BoostFundsAdded',
+    chainflip,
+    (event) =>
+      event.data.boosterId === lp.address &&
+      event.data.boostPool.asset === asset &&
+      event.data.boostPool.tier === boostTier.toString(),
+  );
+
+  // Add funds to the boost pool
+  console.log(`Adding boost funds of ${amount} ${asset} at ${boostTier}bps`);
+  await lpSubmitExtrinsic(
+    chainflip.tx[ingressEgressPalletForChain(chainFromAsset(asset))].addBoostFunds(
+      shortChainFromAsset(asset).toUpperCase(),
+      amountToFineAmount(amount.toString(), assetDecimals(asset)),
+      boostTier,
+    ),
+    lp,
+  );
+
+  return observeBoostFundsAdded;
+}
+
+/// Adds boost funds to the boost pool and does a swap with boosting enabled, then stops boosting and checks the fees collected are correct.
+async function testBoostingForAsset(asset: Asset, boostFee: number, lpUri: string, amount: number) {
+  await using chainflip = await getChainflipApi();
+  console.log(`Testing boosting for ${asset} at ${boostFee}bps`);
+
+  // Start with a clean slate by stopping boosting before the test
+  const preTestStopBoostingEvent = await stopBoosting(asset, boostFee, lpUri, false);
```

Comment:
We can be a bit clearer here. It makes sense for someone to potentially want to run this test with `boostFee: 22`, however, this test will fail immediately since stopBoosting will try to stopBoosting for tier 22 which doesn't exist. We should be able to provide any boostFee to the test (since we can provide any fee to the channels).

Reply:
Now that we are creating default boost pools and filling them with funds, running the test on any tier lager than 5 will break the test because it will use that tier instead. So I thought it would be better to just hard code the test to use tier 4.